WcaOnRails/app/models/competition.rb

```diff
@@ -43,6 +43,12 @@ def must_have_at_least_one_event
       errors.add(:eventSpecs, "Competition must have at least one event")
     end
   end
+  validate :must_have_at_least_one_delegate, if: :isConfirmed?
+  def must_have_at_least_one_delegate
+    if delegate_ids.length == 0
+      errors.add(:delegates, "Competition must have at least one WCA delegate")
```

Comment:
Yeah, events are special, it should be pretty easy to write some code that checks `competition.errors.has_key? "event_specs"`, and shows the error message, though.